services/version.js

```diff
@@ -147,11 +200,31 @@ function slice(v, releaseType) {
   }
 }
 
+/**
+ * Returns the lowest possible version that satisfies a given semver range expression.
+ * For example, rangeStart(“^1.2.3”) would return “1.2.3” because it is the version string of the first comparator in the first set of the range object, which is >=1.2.3 and <2.0.0.
+ *
+ * @param {string} v - The semver range to evaluate
+ * @returns {string|null} The version string of the first comparator in the first set of the range object, or null if the range is not valid
```

Comment:
When does this return `null`? I'd have thought an invalid semver range would throw `TypeError: Invalid SemVer Range` when we try to construct it.

Reply:
this one is still outstanding

Reply:
done

Reply:
updated